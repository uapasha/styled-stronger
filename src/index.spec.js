import 'jest-styled-components';
import { create } from 'react-test-renderer';

import React from 'react';
import styled from 'styled-components';

import stronger, { enhanceStyled } from '.';

const propFunc = ({ float }) => float || 'initial';

describe('stronger', () => {
  it('handles no amplifier', () => {
    const enhancedStyles = stronger(1)`
        background-color: #fff;
        float: ${propFunc};
        font-size: 12px;
    `;

    const ButtonEnhancedLiteral = styled.button(enhancedStyles);
    expect(create(<ButtonEnhancedLiteral float="left" />)).toMatchStyledComponentsSnapshot();
  });

  it('handles string amplifier', () => {
    const enhancedStyles = stronger('dsff')`
        background-color: #fff;
        float: ${propFunc};
        font-size: 12px;
    `;

    const ButtonEnhancedLiteral = styled.button(enhancedStyles);
    expect(create(<ButtonEnhancedLiteral float="left" />)).toMatchStyledComponentsSnapshot();
  });

  it('handles amplified once', () => {
    const enhancedStyles = stronger()`
        background-color: #fff;
        float: ${propFunc};
        font-size: 12px;
    `;

    const ButtonEnhancedLiteral = styled.button(...enhancedStyles);
    expect(create(<ButtonEnhancedLiteral float="left" />)).toMatchStyledComponentsSnapshot();
  });

  it('handles 20 amplifier', () => {
    const enhancedStyles = stronger(20)`
        background-color: #fff;
        float: ${propFunc};
        font-size: 12px;
    `;

    const ButtonEnhancedLiteral = styled.button(enhancedStyles);
    expect(create(<ButtonEnhancedLiteral float="left" />)).toMatchStyledComponentsSnapshot();
  });

  it('handles 5 amplifier and no prop passed', () => {
    const enhancedStyles = stronger(20)`
        background-color: #fff;
        float: ${propFunc};
        font-size: 12px;
    `;

    const ButtonEnhancedLiteral = styled.button(enhancedStyles);
    expect(create(<ButtonEnhancedLiteral />)).toMatchStyledComponentsSnapshot();
  });

  it('applied once and no amplified set are equal', () => {
    const enhancedStyles = stronger()`
        background-color: #fff;
        float: ${propFunc};
        font-size: 12px;
    `;
    const enhancedStyles1 = stronger(1)`
        background-color: #fff;
        float: ${propFunc};
        font-size: 12px;
    `;

    expect(enhancedStyles).toEqual(enhancedStyles1);
  });
});

describe('enhanceStyled', () => {
  it('handles no amplifier', () => {
    const ButtonEnhancedStyled = enhanceStyled(styled.p)`
      background-color: #fff;
      float: ${propFunc};
      font-size: 12px;
    `;
    expect(create(<ButtonEnhancedStyled float="right" />)).toMatchStyledComponentsSnapshot();
  });

  it('handles amplifier of 1', () => {
    const ButtonEnhancedStyled = enhanceStyled(styled.p, 1)`
      background-color: #fff;
      float: ${propFunc};
      font-size: 12px;
    `;
    expect(create(<ButtonEnhancedStyled />)).toMatchStyledComponentsSnapshot();
  });

  it('handles amplifier of 0', () => {
    const ButtonEnhancedStyled = enhanceStyled(styled.p, 0)`
      background-color: #fff;
      float: ${propFunc};
      font-size: 12px;
    `;
    expect(create(<ButtonEnhancedStyled />)).toMatchStyledComponentsSnapshot();
  });

  it('handles amplifier of 100', () => {
    const ButtonEnhancedStyled = enhanceStyled(styled.p, 100)`
      background-color: #fff;
      float: ${propFunc};
      font-size: 12px;
    `;
    expect(create(<ButtonEnhancedStyled />)).toMatchStyledComponentsSnapshot();
  });
});
