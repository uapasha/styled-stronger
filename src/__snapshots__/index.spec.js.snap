// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enhanceStyled handles amplifier of 0 1`] = `
.ixNkzK {
  background-color: #fff;
  float: initial;
  font-size: 12px;
}

<p
  className="sc-gzVnrw ixNkzK"
/>
`;

exports[`enhanceStyled handles amplifier of 1 1`] = `
.feubwF {
  background-color: #fff;
  float: initial;
  font-size: 12px;
}

<p
  className="sc-bZQynM feubwF"
/>
`;

exports[`enhanceStyled handles amplifier of 100 1`] = `


<p
  className="sc-htoDjs icXvAP"
/>
`;

exports[`enhanceStyled handles no amplifier 1`] = `
.gsSXuA {
  background-color: #fff;
  float: right;
  font-size: 12px;
}

<p
  className="sc-EHOje gsSXuA"
/>
`;

exports[`stronger handles 5 amplifier and no prop passed 1`] = `


<button
  className="sc-ifAKCX jtQzFX"
/>
`;

exports[`stronger handles 20 amplifier 1`] = `


<button
  className="sc-bxivhb hjHcYL"
/>
`;

exports[`stronger handles amplified once 1`] = `
.eSgojR {
  background-color: #fff;
  float: left;
  font-size: 12px;
}

<button
  className="sc-htpNat eSgojR"
/>
`;

exports[`stronger handles no amplifier 1`] = `
.gtlzod {
  background-color: #fff;
  float: ;
  font-size: 12px;
}

<button
  className="sc-bdVaJa gtlzod"
/>
`;

exports[`stronger handles string amplifier 1`] = `
.kCSSUt {
  background-color: #fff;
  float: ;
  font-size: 12px;
}

<button
  className="sc-bwzfXH kCSSUt"
/>
`;
